.github/workflows/test.yaml

```diff
@@ -31,8 +31,18 @@ jobs:
       - uses: actions/setup-go@v2
         with: { go-version: 1.x }
       - run: go mod download
-      - env: { KUBERNETES: "${{ matrix.kubernetes }}" }
-        run: ENVTEST_K8S_VERSION="${KUBERNETES#default}" make check-envtest
+      - run: ENVTEST_K8S_VERSION="${KUBERNETES#default}" make check-envtest
+        env:
+          KUBERNETES: "${{ matrix.kubernetes }}"
+          GO_TEST: go test --coverprofile 'envtest.coverage' --coverpkg ./internal/...
+
+      # Upload coverage to GitHub
+      - run: gzip envtest.coverage
+      - uses: actions/upload-artifact@v2
+        with:
+          name: "kubernetes-api=${{ matrix.kubernetes }}"
+          path: envtest.coverage.gz
+          retention-days: 1
```

Comment:
These coverage profiles are large and only uploaded as artifacts so the final job can download them to generate reports. The minimum possible retention is one day.